exotations/task_maps/task_map/src/JointAccBackFiniteDiff.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *      Author: Christopher E. Mower
+ *
+ * Copyright (c) 2018, University Of Edinburgh
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of  nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "task_map/JointAccBackFiniteDiff.h"
+
+REGISTER_TASKMAP_TYPE("JointAccBackFiniteDiff", exotica::JointAccBackFiniteDiff);
+
+namespace exotica {
+
+  JointAccBackFiniteDiff::JointAccBackFiniteDiff(){
+  }
+
+  JointAccBackFiniteDiff::~JointAccBackFiniteDiff(){
+  }
+
+  void JointAccBackFiniteDiff::assignScene(Scene_ptr scene) {
+    scene_ = scene;
+    Initialize();
+  }
+
+  void JointAccBackFiniteDiff::Instantiate(JointAccBackFiniteDiffInitializer& init) {
+    init_ = init;
+  }
+
+  void JointAccBackFiniteDiff::Initialize() {
+    joint_state_prev_1_ = init_.JointState_prev_1;
+    joint_state_prev_2_ = init_.JointState_prev_2;
+    N = scene_->getSolver().getNumControlledJoints();
+    I_ = Eigen::MatrixXd::Identity(N,N);
+  }
+
+  void JointAccBackFiniteDiff::setPrevJointState(Eigen::VectorXd joint_state) {
+    joint_state_prev_2_ = joint_state_prev_1_;
+    joint_state_prev_1_ = joint_state;
+  }
+  
+  void JointAccBackFiniteDiff::update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi) {
+    phi = x - 2*joint_state_prev_1_ + joint_state_prev_2_;
```

Comment:
This cost will be dependent on delta time. If you use the same formula at different control rate, the results will be different.

Reply:
... and dt (tau) is not available for any problem/scene apart from the time-indexed problems. Henceforth it should be an initialiser variable.

Reply:
So change should be as follows?
* In xml `<JointAccBackFiniteDiff Name="JointAcc" Hz="100.00">`
* In task map `phi = Hz_*(x - 2*joint_state_prev_1_ + joint_state_prev_2_);` since `Hz = 1/dt`

and similar changes for Vel and Jerk?

Reply:
By adding this a new issue arises, see [here](https://youtu.be/tY4CdS28YJw). I will *not* include this suggestion in next commit.